test/e2e/network/dual_stack.go

```diff
@@ -0,0 +1,308 @@
+		framework.ExpectNoError(err, "failed to delete pod")
+	})
+
+	// takes close to 140s to complete, so doesn't need to be marked [SLOW]
+	ginkgo.It("should be able to reach pod on ipv4 and ipv6 ip", func() {
+		nginxDeploymentName := "dualstack-server-nginx"
+		networkUtilsDeploymentName := "dualstack-client-utils"
+
+		// get all schedulable nodes to determine the number of replicas for pods
+		// this is to ensure connectivity from all nodes on cluster
+		nodeList := framework.GetReadySchedulableNodesOrDie(cs)
+		gomega.Expect(nodeList).NotTo(gomega.BeNil())
+
+		if len(nodeList.Items) < 1 {
+			framework.Failf("Expect at least 1 node, got %v", len(nodeList.Items))
+		}
+
+		replicas := int32(len(nodeList.Items))
+
+		// to ensure all the pods land on different nodes and we can thereby
+		// validate connectivity across all nodes.
+		serverDeploymentSpec := &apps.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   nginxDeploymentName,
+				Labels: map[string]string{"test": "dualstack-test-server"},
+			},
+			Spec: apps.DeploymentSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"test": "dualstack-test-server"}},
+				Strategy: apps.DeploymentStrategy{
+					Type: apps.RollingUpdateDeploymentStrategyType,
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"test": "dualstack-test-server"},
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:  "dualstack-test-server",
+								Image: nginxImageWithIPv6,
+							},
+						},
+						Affinity: &v1.Affinity{
+							PodAntiAffinity: &v1.PodAntiAffinity{
+								RequiredDuringSchedulingIgnoredDuringExecution: []v1.PodAffinityTerm{
+									{
+										LabelSelector: &metav1.LabelSelector{
+											MatchExpressions: []metav1.LabelSelectorRequirement{
+												{
+													Key:      "test",
+													Operator: metav1.LabelSelectorOpIn,
+													Values:   []string{"dualstack-test-server"},
+												},
+											},
+										},
+										TopologyKey: "kubernetes.io/hostname",
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		}
+
+		clientDeploymentSpec := &apps.Deployment{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:   networkUtilsDeploymentName,
+				Labels: map[string]string{"test": "dualstack-test-client"},
+			},
+			Spec: apps.DeploymentSpec{
+				Replicas: &replicas,
+				Selector: &metav1.LabelSelector{MatchLabels: map[string]string{"test": "dualstack-test-client"}},
+				Strategy: apps.DeploymentStrategy{
+					Type: apps.RollingUpdateDeploymentStrategyType,
+				},
+				Template: v1.PodTemplateSpec{
+					ObjectMeta: metav1.ObjectMeta{
+						Labels: map[string]string{"test": "dualstack-test-client"},
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{
+							{
+								Name:    "dualstack-test-client",
+								Image:   networkUtilImage,
```

Comment:
There is already an image for this https://github.com/kubernetes/kubernetes/tree/master/test/images/hostexec

We can use it `Image:           imageutils.GetE2EImage(imageutils.Hostexec),`